<launch>
    <arg name="use_sim" default="true" />
    <arg name="use_gripper" default="true" />
    <arg name="sim_start_pose" default="pose_xy0" />
    <!-- <arg name="controller" value="cartesian_impedance_controller_damping_ratio" />  -->
    <arg name="controller" value="trajectory_cartesian_impedance_controller" />  
    <arg name="robot_ip" default="franka" />  <!-- Aggiunto -->
    <arg name="arm_id" default="fr3" />  <!-- Aggiunto -->
    <arg name="record" default="false" />

    <group if="$(arg use_sim)">
        <include file="$(find nist_cables)/launch/sim.launch" pass_all_args="true"> 
        <arg name="robot_ip" value="franka" />
        <arg name="arm_id" value="fr3" />
        <arg name="robot" value="fr3" />
        <arg name="use_gripper" value="$(arg use_gripper)" />
        <arg name="controller" value="$(arg controller)" />
        <arg name="start_pose" value="$(arg sim_start_pose)" />
        <arg name="interactive_marker" value="false" />
        </include>
    </group>

    <!-- Control real robot --> 
    <group unless="$(arg use_sim)">
        <include file="$(find nist_cables)/launch/real.launch" pass_all_args="true">
        <arg name="robot_ip" value="franka" />
        <arg name="arm_id" value="fr3" />
        <arg name="robot" value="fr3" />
        <arg name="load_gripper" value="true" />
        <arg name="launch-prefix" value="sleep 5" />
        <arg name="controller" value="$(arg controller)" />
        <arg name="use_sim_time" value="false" />
        </include>
    </group>


    <!-- Conditional USB Camera Bringup and Rosbag Record -->
    <group if="$(arg record)">
        <node name="rosbag_record" pkg="rosbag" type="record"
        args="-o $(find dhb_ros)/data/new/relay/
        /franka_state_controller/franka_states
        /franka_state_controller/joint_states
        /franka_state_controller/F_ext
        /franka_state_controller/F_ext_base
        /ft_sensor/ft_compensated
        /ft_sensor/ft_compensated_base
        /franka_gripper/joint_states
        /cartesian_impedance_controller_damping_ratio/equilibrium_pose
        /usb_cam/image_raw/compressed"
        output="screen"/>
    </group>

    <include file="$(find dlo_detection)/launch/detection_pipeline.launch" />

    <node name="motion_AS" pkg="dhb_ros" type="dhb_action_server.py" output="screen">
        <param name="name" value="DHB_motion" />
    </node>





</launch>